woof-code/rootfs-skeleton/usr/sbin/sfs_load

```diff
@@ -49,7 +49,13 @@
 
 . /etc/rc.d/PUPSTATE
 
-if [ -L /bin -o -L /sbin -o -L /lib -o -L /lib64 -o -L /lib32 -o -L /libx32 ]; then
+sym_list=""
+for fld2 in bin sbin lib lib64 lib32 libx32
+do
+  [ -L /$fld2 ] && sym_list="$sym_list $fld2"
+done
+
+if [ "$sym_list" != "" ]; then
```

Comment:
`if [ -n "$sym_list" ]; then`

Reply:
Replying to all 7 comments made over a very short period of time.........
1. libx32 as a symlink with no libx32 directory in sfs works fine (tested)
2. the test is "-a" - both directory must exist AND it isn't a symlink in the sfs
3. the style changes are not explained/justified - why are they needed?
4. moving the "if" is a different additional update
5. the error message needs to provide information on why the sfs load is failing - if it can't be translatable it will need to be just English
6. I prefer to leave it to the user to decide whether they want to load an sfs having provided them with information - don't like things that hard fail with no get-out clause

Reply:
> libx32 as a symlink with no libx32 directory in sfs works fine (tested)

The logical bug happens when the main SFS has a directory at /libx32 (or any other directory in the list), while the devx doesn't have anything there.

> the test is "-a" - both directory must exist AND it isn't a symlink in the sfs

Enjoy this counter-example:

```
/tmp$ mkdir x
/tmp$ rm y
rm: cannot remove 'y': No such file or directory
/tmp$ [ -d x -a ! -L y ] && echo bug
bug
```

> the style changes are not explained/justified - why are they needed?

I wouldn't say they're absolutely _needed_, it's just cleaner, more efficient in some cases, and more consistent with non-Puppy shell scripts.

> the error message needs to provide information on why the sfs load is failing - if it can't be translatable it will need to be just English

How about changing this:

```
$(gettext "The directory $FLD_INVALID in this sfs will overwrite the system symlink. This may cause problems. Do you want to continue?")
```

to this?

```
$(gettext "The following directories this sfs will overwrite system symlinks. This may cause problems. Do you want to continue?")

-- list of directories goes here --
```

> I prefer to leave it to the user to decide whether they want to load an sfs having provided them with information - don't like things that hard fail with no get-out clause

There are cases where a SFS is **guaranteed** to break the system. There's no reason to allow this. People who want to ensure the destruction of their OS can just `rm -rf /; kill -9 1` or something.

Reply:
> I prefer to leave it to the user to decide whether they want to load an sfs having provided them with information - don't like things that hard fail with no get-out clause

Lowest common denominator? Imagine if traffic rules were legislated by your rationale.

Reply:
`<hypothetical>`
APP: If you proceed all your funds run the risk of being transferred to a Mafia cartel. Do you wish to proceed?
USER: ** hits wrong button **
APP: **I'm fatter now**
`</hypothetical>`

Who is responsible?